### PR TITLE
Binyamin/support magic methods

### PR DESCRIPTION
**Pull request by binyamin555**

### CODE REVIEW

**Comment by binyamin555** [comment]
Maybe we should implement the rest of the methods before merging PR?

File: test.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+def bar(x, y):
+    return x + y
+
+
+extend_type_with(function, FunctionExtension)
+
+
+print(foo @ bar)
+print((foo @ bar)(1, 2))  # for some reason, this also print None.
```
Comment (elazarcoh) [inline test.py:46]: because print returns None

**Review by elazarcoh — APPROVED** [review]
seems good. didnt run yet

File: test.py (hunk 1 above)
Comment (binyamin555) [reply test.py:46]: Oh, right! All of this function mess was very confusing xD.

**Comment by elazarcoh** [comment]
updated the build.
now you can use `hatch run build` and it will use the same environment for the build

**Comment by binyamin555** [comment]
@elazarcoh I implemented the other protocols. I think we should wait with the async and buffer protocols for now.

I also can't seem to fix reverse methods (e.g. `__radd__`), I probably need to implement them in the `impl__xxx__` functions manually.

Should we merge them anyway and add support for reverse methods and the other protocols in another PR / version?

**Comment by elazarcoh** [comment]
Yeah that's fine, we can merge now. Let's add a TODO list to the README

**Pull request merged**